### PR TITLE
feat: Infracost template file support

### PR DESCRIPTION
**Pull request by hugorut**
Adds support for Infracost template files using go text/templates.

### CODE REVIEW

File: cmd/infracost/generate.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		ValidArgs: []string{"--", "-"},
+		RunE:      gen.run,
+	}
+
+	cmd.Flags().StringVar(&gen.projectPath, "project-path", ".", "Path to the Terraform project you want to run the template file on")
+	cmd.Flags().StringVar(&gen.templatePath, "template-path", "", "Path to the Infracost template file which will generate the yml output")
+	cmd.Flags().StringVar(&gen.outFile, "out-file", "", "Path to the Infracost template file which will generate the yml output")
+
+	return cmd
+}
```
Comment (aliscott) [inline cmd/infracost/generate.go:38]: @alikhajeh1 would be good if you review this bit as well.

File: cmd/infracost/generate.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+		Short: "Generate Infracost YML file from a template file",
+		Long:  "Generate Infracost YML file from a template file",
+		Example: `
+      infracost generate config --project-path . --template-path infracost.yml.tmpl
+      `,
+		ValidArgs: []string{"--", "-"},
+		RunE:      gen.run,
+	}
+
+	cmd.Flags().StringVar(&gen.projectPath, "project-path", ".", "Path to the Terraform project you want to run the template file on")
```
Comment (aliscott) [inline cmd/infracost/generate.go:33]: Not sure if `project` is the right term here. We just use `--path` elsewhere to refer to the repository path, but understand we might want to make this more explicit. Any ideas @alikhajeh1?

**Review by aliscott — APPROVED** [review]

File: cmd/infracost/generate.go (hunk 2 above)
Comment (alikhajeh1) [reply cmd/infracost/generate.go:33]: @aliscott and @hugorut yeah `project` is confusing here as we use that term to mean another thing in Infracost.
Maybe `--repo-path` with `Path to the Terraform repo or directory you want to run the template file on`

File: cmd/infracost/generate.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+		Long:  "Generate Infracost config file from a template file",
+		Example: `
+      infracost generate config --project-path . --template-path infracost.yml.tmpl
+      `,
+		ValidArgs: []string{"--", "-"},
+		RunE:      gen.run,
+	}
+
+	cmd.Flags().StringVar(&gen.projectPath, "project-path", ".", "Path to the Terraform project you want to run the template file on")
+	cmd.Flags().StringVar(&gen.templatePath, "template-path", "", "Path to the Infracost template file which will generate the yml output")
```
Comment (alikhajeh1) [inline cmd/infracost/generate.go:34]: @hugorut maybe `Path to the Infracost template file that will generate the config-file yaml output` ?

**Pull request merged**